include/alpaka/atomic/AtomicAtomicRef.hpp

```diff
@@ -219,7 +219,23 @@ namespace alpaka
                 T result;
                 do
                 {
+#    if BOOST_COMP_GNUC
+#        pragma GCC diagnostic push
```

Comment:
`GCC` is understood by _clang_ compiler hence this would work (similar to usage at /alpaka/offset/Traits.hpp:42):

```
#if BOOST_COMP_CLANG || BOOST_COMP_GNUC
#    pragma GCC diagnostic push
#    pragma GCC diagnostic ignored "-Wfloat_equal"
#endif
         result = ((old == compare) ? value : old);
#if BOOST_COMP_CLANG || BOOST_COMP_GNUC
#    pragma GCC diagnostic pop
#endif
```

Reply:
I see, thanks for the suggestion !